File: longest-consecutive-one/consecutive.py
from typing import List

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        longest = count = 0
        for num in nums:
            if num != 1:
                count = 0
                continue
            count += 1
            longest = max(longest, count)
        return longest

'''

my approach
***********
1. initialize longest and count with 0
2. iterate through nums and if a 1 is reached;
3. increment count and assign longest the greater value of max and longest
4. else, reset count to 0 and skip the iteration entirely
'''

File: longest-consecutive-one/Consecutive.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Consecutive{

    public static void main(String[] args) {
        List<Integer> nums = new ArrayList<>(Arrays.asList(1, 2, 3, 2, 1, 1, 1, 1, 1, 3, 4, 1, 2, 1, 1, 1, 2));
        System.out.println(maxConsecutiveOnes(nums));
    }
    
    public static int maxConsecutiveOnes(List<Integer> nums ) {
        int longest = 0, count = 0;
        for (Integer num : nums) {
            if (num != 1) {
                count = 0;
                continue;
            }
            count += 1;
            longest = Math.max(longest, count);
        }
        return longest;
    }
}